torch/lib/TensorLib/CUDAGenerator.cpp

```diff
@@ -0,0 +1,44 @@
+#ifdef TENSORLIB_CUDA_ENABLED
+
+#include "TensorLib/CUDAGenerator.h"
+#include "TensorLib/Context.h"
+#include <stdexcept>
+
+#define const_generator_cast(generator) \
+  dynamic_cast<const CUDAGenerator&>(generator)
+
+namespace tlib {
+
+CUDAGenerator::CUDAGenerator(Context * context_)
```

Comment:
I agree, they are copied verbatim from THPP. I think the idea of having a generator object is good but the implementation is not complete on CUDA yet. Better to have the API be what we want it eventually to be rather than force a change later.